flowpipe/graph.py

```diff
@@ -278,13 +370,21 @@ def serialize(self):  # pragma: no cover
 
         return self._serialize()
 
-    def _serialize(self):
-        """Serialize the graph in it's grid form."""
+    def _serialize(self, with_subgraphs=True):
+        """Serialize the graph in it's grid form.
+
+        Args:
+            with_subgraphs (bool): Set to false to avoid infinite recursion
+        """
         data = OrderedDict(
             module=self.__module__,
             cls=self.__class__.__name__,
             name=self.name)
         data['nodes'] = [node.to_json() for node in self.nodes]
+        if with_subgraphs:
+            data['subgraphs'] = [
+                graph._serialize(with_subgraphs=False)
+                for graph in self.subgraphs.values()]
         return data
```

Comment:
I notice the `with_subgraphs` argument is set to `False` when iterating over subgraphs. Does this mean subgraphs can't have subgraphs? I think I previously used subgraphs within subgraphs in real world scenario's so this would be a hindrance. Can we not have a property `has_subgraph` for a graph? This would also eliminate the need for the `with_subgraphs` argument in this method.

Reply:
Subgraphs are gathered by following all connections of all nodes and collecting their graphs along the way. This is performed both upstream and downstream so any subgraph of a graph will in turn contain that main graph as it's subgraph. This also means that any graph also contains all subgraphs of all subgraphs all the way down (and up). Therefore we need the `with_subgraphs` variable to avoid an infinite loop here.  Does that male any sense?